homeassistant/components/icloud/__init__.py

```diff
@@ -1 +1,629 @@
+ATTR_ACCOUNTNAME = 'account_name'
+ATTR_BATTERY = 'battery'
+ATTR_BATTERYSTATUS = 'battery_status'
+ATTR_DEVICENAME = 'device_name'
+ATTR_DEVICESTATUS = 'device_status'
+ATTR_LOWPOWERMODE = 'low_power_mode'
+ATTR_OWNERNAME = 'owner_fullname'
+
+DEVICE_STATUS_SET = ['features', 'maxMsgChar', 'darkWake', 'fmlyShare',
+                     'deviceStatus', 'remoteLock', 'activationLocked',
+                     'deviceClass', 'id', 'deviceModel', 'rawDeviceModel',
+                     'passcodeLength', 'canWipeAfterLock', 'trackingInfo',
+                     'location', 'msg', 'batteryLevel', 'remoteWipe',
+                     'thisDevice', 'snd', 'prsId', 'wipeInProgress',
+                     'lowPowerMode', 'lostModeEnabled', 'isLocating',
+                     'lostModeCapable', 'mesg', 'name', 'batteryStatus',
+                     'lockedTimestamp', 'lostTimestamp', 'locationCapable',
+                     'deviceDisplayName', 'lostDevice', 'deviceColor',
+                     'wipedTimestamp', 'modelDisplayName', 'locationEnabled',
+                     'isMac', 'locFoundEnabled']
+
+DEVICE_STATUS_CODES = {
+    '200': 'online',
+    '201': 'offline',
+    '203': 'pending',
+    '204': 'unregistered',
+}
+
+_CONFIGURING = {}
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVICE_ICLOUD_PLAY_SOUND = 'play_sound'
+SERVICE_ICLOUD_DISPLAY_MESSAGE = 'display_message'
+SERVICE_ICLOUD_LOST_DEVICE = 'lost_device'
+SERVICE_ICLOUD_UPDATE = 'update'
+SERVICE_ICLOUD_RESET = 'reset'
+SERVICE_ATTR_LOST_DEVICE_MESSAGE = 'message'
+SERVICE_ATTR_LOST_DEVICE_NUMBER = 'number'
+SERVICE_ATTR_LOST_DEVICE_SOUND = 'sound'
+
+SERVICE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ACCOUNTNAME): cv.string,
+})
+
+SERVICE_SCHEMA_PLAY_SOUND = vol.Schema({
+    vol.Required(ATTR_ACCOUNTNAME): cv.string,
+    vol.Required(ATTR_DEVICENAME): cv.string,
+})
+
+SERVICE_SCHEMA_DISPLAY_MESSAGE = vol.Schema({
+    vol.Required(ATTR_ACCOUNTNAME): cv.string,
+    vol.Required(ATTR_DEVICENAME): cv.string,
+    vol.Required(SERVICE_ATTR_LOST_DEVICE_MESSAGE): cv.string,
+    vol.Optional(SERVICE_ATTR_LOST_DEVICE_SOUND): cv.boolean,
+})
+
+SERVICE_SCHEMA_LOST_DEVICE = vol.Schema({
+    vol.Required(ATTR_ACCOUNTNAME): cv.string,
+    vol.Required(ATTR_DEVICENAME): cv.string,
+    vol.Required(SERVICE_ATTR_LOST_DEVICE_NUMBER): cv.string,
+    vol.Required(SERVICE_ATTR_LOST_DEVICE_MESSAGE): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(ATTR_ACCOUNTNAME): cv.slugify,
+        vol.Optional(CONF_MAX_INTERVAL, default=30): cv.positive_int,
+        vol.Optional(CONF_GPS_ACCURACY_THRESHOLD, default=500): cv.positive_int
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+ICLOUD_COMPONENTS = [
+    'sensor', 'device_tracker'
+]
+
+
+def setup(hass, config):
+    """Set up the iCloud component."""
+    async def play_sound(service):
```

Comment:
Why are the service handlers coroutines? Does the api support asyncio?

Reply:
Emmm, nope, my bad.

Removing `async`s